### PR TITLE
Josh demo and some fixes

### PR DESCRIPTION
**Pull request by jcorbin**
Demo includes:
- streaming echo
- streaming grep -n analog
- streaming node repl
- streaming arbitrary program execution (best idea ever...)

### CODE REVIEW

File: node/examples/josh_test_client.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
```
Comment (mmihic) [inline node/examples/josh_test_client.js]: josh_test_client?

File: node/examples/josh_test_server.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+grepn.canStream = true;
+chan.handler.register('grepn', grepn);
+
+chan.listen(4040, '127.0.0.1');
+
+var spawn = require('child_pty').spawn;
+// var spawn = require('child_process').spawn;
+
+function exec(req, res) {
```
Comment (mmihic) [inline node/examples/josh_test_server.js]: this is awesome

**Comment by kriskowal** [comment]
:ok:

File: node/examples/josh_test_client.js (hunk 1 above)
Comment (jcorbin) [inline node/examples/josh_test_client.js]: yeah... I felt like it was the best token to stand in for "these 4 random things I ended up with in my demo" ;-)

I guess I could sort out some weird acronym of Echo Exec Repl GrepN ;-)

**Pull request merged**